src/SpecFile_radiacode.cpp

```diff
@@ -520,18 +520,49 @@ bool SpecFile::load_from_radiacode_spectrogram( std::istream& input )
       
       // We'll be pretty generous about allowing invalid lines
       if( (skipped_lines > 5) && (total_lines > 10) && (skipped_lines > (total_lines/10)) )
-        throw runtime_error( "To many invalid lines" );
-      
+        throw runtime_error("Too many invalid lines");
+
       trim( line );
       if( line.empty() )
       {
         skipped_lines += 1;
         continue;
       }
-      
-      // The second line in the file starts with "Spectrum:", and looks like its probably an icon
-      //  of the spectrum, and an example one I looked at was 12 kb, we'll skip it
-      
+
+      // The second line in the file - "Spectrum:" - is a hex-encoded
+      // representation of the total recorded dose since the last device reset.
+      // 0-3: spectrum accumulation time in seconds (uint32)
+      // 4-15: calibration factors a0, a1, a2 (float[3])
+      // 16 .. : counts per channel (uint32[1024])
+
+      string pfx = "Spectrum: ";  // the space is intentional
+      if ((string::npos != line.find(pfx)) && (1 == total_lines)) {
+        try {
+          uint8_t raw_bytes[16];
+          vector<float> cal_coefs;
+          int ds = pfx.length();
+
+          // "unhexlify"
+          for (int i = 0; i < sizeof(raw_bytes); i++) {
+            string tmp = line.substr(ds + i * 3, 2);
```

Comment:
I think if `line` is shorter than 27 bytes, than there could be problems with this substr.

Reply:
One more length check, comin' right up.